app/commands/solution/generate_readme_file.rb

```diff
@@ -0,0 +1,81 @@
+class Solution
+  class GenerateReadmeFile
+    include Mandate
+
+    initialize_with :solution
+
+    def call
+      [preamble, introduction, instructions, source].compact_blank.join("\n\n")
+    end
+
+    private
+    def preamble
+      if solution.git_exercise.hints.present?
+        hints_text = "If you get stuck on the exercise, check out `HINTS.md`, but try and solve it without using those first :)" # rubocop:disable Layout/LineLength
+      end
+
+      "# #{solution.exercise.title}
+
+Welcome to #{solution.exercise.title} on Exercism's #{solution.track.title} Track.
+If you need help running the tests or submitting your code, check out `HELP.md`.
+#{hints_text}".strip
+    end
+
+    def introduction
+      return unless solution.git_exercise.introduction
```

Comment:
```suggestion
      return unless solution.git_exercise.introduction.present?
```

(I could see tracks having blank files)